opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -746,7 +747,25 @@ namespace Opm {
 
         well_container_generic_.clear();
         for (auto& w : well_container_)
-          well_container_generic_.push_back(w.get());
+            well_container_generic_.push_back(w.get());
+
+        const auto& network = schedule()[time_step].network();
```

Comment:
This is for setting the dynamic THP constraint when creating the wells in well containers. Because we have the `dynamic_thp_limit_` in WellInterface, we need to initialize it based on the network nodal pressures. 

BlackoilWellModelGeneric::updateNetworkPressures() is for updating the THP constraint during the iterative updating of network nodal pressures and also thp value in the well state if it is under THP control. 

They are for different purposes.